tests/test_csc.py

```diff
@@ -151,6 +156,64 @@ def setUp(self):
             self.assertEqual(alarm.maxSeverity, AlarmSeverity.NONE)
             self.assertFalse(alarm.acknowledged)
 
+    async def test_mute(self):
+        """Test the mute and unmute command."""
+        async with Harness(config_dir=TEST_CONFIG_DIR) as harness:
+            await salobj.set_summary_state(harness.remote, state=salobj.State.ENABLED,
+                                           settingsToApply="enabled.yaml")
+            nrules = len(harness.csc.model.rules)
+
+            user1 = "test_mute 1"
+            # mute all alarms for a short time,
+            # then wait for them to unmute themselves
+            await harness.remote.cmd_mute.set_start(name="Enabled.*",
+                                                    duration=0.1,
+                                                    severity=AlarmSeverity.SERIOUS,
+                                                    mutedBy=user1,
+                                                    timeout=STD_TIMEOUT)
+            muted_names = []
+            while len(muted_names) < nrules:
+                data = await harness.remote.evt_alarm.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(data.mutedSeverity, AlarmSeverity.SERIOUS)
+                self.assertEqual(data.mutedBy, user1)
+                muted_names.append(data.name)
+            self.assertEqual(set(muted_names), set(harness.csc.model.rules))
+
+            unmuted_names = []
+            while len(unmuted_names) < nrules:
+                data = await harness.remote.evt_alarm.next(flush=False, timeout=STD_TIMEOUT)
```

Comment:
I feel like this block, and the one above it, could be turned into a single method "check_mutedness" or something, that takes a severity and a user and does all this checking. Because unless I'm misreading, they're otherwise identical.

Reply:
I prefer the current code so I can see exactly what is happening without having to hunt up the definition of another method.

I considered an `assertAlarmEvent` method but it would need a lot of keyword arguments (or a kwargs) and it's a nuisance for boolean fields because I don't want to test those for equality.